src/runway/path.py

```diff
@@ -0,0 +1,130 @@
+"""Runway configuration 'path' settings."""
```

Comment:
Can we add documentation here to be pulled in by `automodule`? It can then be added to `docs/source/runway_config.rst` under the `Module` header unless you can think of a better place to put it.

May also be worth adding a link in the docs on `runway.config.ModuleDefinition` where it talks about `path`.

The docs here should touch on local paths (`./` and `sampleapp.cfn`) - can likely move any in-depth path details from `runway.config.ModuleDefinition` to here. it should also mention _remote sources_ and link to them. There should be examples of the different ways to define `path`